modules/aioseop_sitemap.php

```diff
@@ -1514,7 +1515,22 @@ public function sitemap_output_hook( $query ) {
 				if ( $gzipped ) {
 					ob_start();
 				}
-				$this->do_rewrite_sitemap( $sitemap_type, $page );
+
+				$content = $this->do_rewrite_sitemap( $sitemap_type, $page );
+
+				// if the sitemap has no content, it's probabaly invalid and is being called directly.
+				// @issue https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/2190
+				if ( empty( $content ) ) {
+					$query->set_404();
+					status_header( 404 );
+					header( "Content-Type: text/html; charset=$blog_charset", true );
+					nocache_headers();
+					include( get_404_template() );
```

Comment:
Would it be better to use `get_template_part()`? 

https://developer.wordpress.org/reference/functions/get_template_part/

Reply:
this is the recommended method